content/en/docs/Tutorials/basic.md

```diff
@@ -0,0 +1,242 @@
+used with Baker because:
+
+* it includes a header row
+* some fields have values with commas and thus are enclosed in double-quotes. Baker doesn't support it
+* the file is uncompressed
+
+For the purpose of this tutorial we've already prepared the final file for you and **it is
+available for downloading [here](/tutorial-base-dataset.csv.gz).**
+{{% /alert %}}
+
+## The required components
+
+* [`List`](/docs/components/inputs/list/): reads the input file from disk
+* [`NotNull`](/docs/components/filters/notnull/): discards all ramens without a top-ten entry
+* [`FileWriter`](/docs/components/outputs/filewriter/): saves the resulting file to disk
+* [`S3`](/docs/components/uploads/s3/): uploads the file to S3
+
+## Baker configuration
+
+An essential thing to do is to create a configuration file for Baker, in
+[TOML](https://github.com/toml-lang/toml) format, selecting the aforementioned components:
+
+```toml
+[input]
+name = "List"
+
+    [input.config]
+    Files = ["/tmp/db.csv.gz"] # put the file wherever you like
+
+[[filter]]
+name = "NotNull"
+
+    [filter.config]
+    Fields = ["top_ten"] # discard all records with an empty top_ten field
+
+[output]
+name = "FileWriter"
+procs = 1 # With our PathString, FileWriter doesn't support concurrency
+fields = ["country"]
+
+    [output.config]
+    PathString = "/tmp/out/{{.Field0}}/ramens.csv.gz"
+
+[upload]
+name="S3"
+
+    [upload.config]
+    Region = "us-east-1"
+    Bucket = "myBucket"
+    Prefix = "ramens/"
+    StagingPath = "/tmp/staging/"
+    SourceBasePath = "/tmp/out/"
+    Interval = "60s"
+    ExitOnError = true
+```
+
+## Create the program
+
+Baker is a Go library. To use it, it is required to create a Go `main()` function,
+[define a `baker.Components`](/docs/how-tos/baker_components/) object and pass it to
+[`baker.MainCLI()`](https://pkg.go.dev/github.com/AdRoll/baker#MainCLI):
+
+```go
+package main
+
+import (
+	"log"
+
+    "github.com/AdRoll/baker"
+)
+
+func main() {
+    components := baker.Components{/* define components */}
+    if err := baker.MainCLI(components); err != nil {
+		log.Fatal(err)
+	}
+}
+```
+
+### Define baker.Components
+
+Our `baker.Components` implementation must:
+
+* include the components we need to use
+* define `FieldByName` and `FieldName` to map our dataset
+
+The simplest way to add the components to Baker is just to add all available components:
+
+```go
+components := baker.Components{
+    Inputs:      input.All,
+    Filters:     filter.All,
+    Outputs:     output.All,
+    Uploads:     upload.All,
+    /* ... */
+}
+```
+
+As for the mapping functions, we need to map from name to `FieldIndex` and vice-versa:
+
+```go
+var fields = []string{
+	"review_num",
+	"brand",
+	"variety",
+	"style",
+	"country",
+	"stars",
+	"top_ten",
+}
+
+var fieldsByName = map[string]baker.FieldIndex{
+	"review_num": 0,
+	"brand":      1,
+	"variety":    2,
+	"style":      3,
+	"country":    4,
+	"stars":      5,
+	"top_ten":    6,
+}
+
```

Comment:
maybe we should use the new `[fields]` section for that?

Reply:
done